### PR TITLE
feat: Add --no-withdrawl flag for "canister delete" for when the canister is out of cycles

### PR DESCRIPTION
**Pull request by jplevyak**
Note: the reject enum is too small (ReplicaError is all we get) to differentiate when a call fails because
the canister is out of cycycles.

If instead we passed from the ic repo: rs/types/error_types/src/lib.rs ErrorCode::OutOfCycles (code 501) back
we could use that to handle that case and automatically fallback to a simple delete.

Signed-off-by: John Plevyak <jplevyak@gmail.com>

### CODE REVIEW

**Review by ericswanson-dfinity — CHANGES_REQUESTED** [review]
Let's start by fixing a spelling error in the new argument name

File: src/dfx/src/commands/canister/delete.rs [hunk 1]
```diff
@@ -43,24 +43,38 @@ pub struct CanisterDeleteOpts {
     #[clap(long, required_unless_present("canister"))]
     all: bool,
 
-    /// Withdraw cycles from canister(s) to canister/wallet before deleting.
+    /// Do not withdrawl cycles, just delete the canister.
     #[clap(long)]
+    no_withdrawl: bool,
```
[the remark this answers is not in the document]
Comment (jplevyak) [reply src/dfx/src/commands/canister/delete.rs]: done and below

**Comment by ericswanson-dfinity** [comment]
Would you please add an entry to the [changelog](https://github.com/dfinity/sdk/blob/master/CHANGELOG.adoc)?

**Review by ericswanson-dfinity — APPROVED** [review]

**Comment by jplevyak** [comment]
Added changelog entry.

**Pull request merged**